.github/workflows/main.yml

```diff
@@ -0,0 +1,71 @@
+# The following configuration is used to setup GitHub Actions
+# for building (and running unit tests) on pushes and PRs.
+# At the moment, we build on two runner types: Linux and MacOS,
+# specific version information can be found further below.
+name: Build On Changes
+
+# run action on all pushes and pull_request for all branches
+on: [push, pull_request]
```

Comment:
Wouldn't `pull_request` be enough?  I mean, as all the changes come via pull requests, it would be unnecessary to run tests on the actual pushes to the repo.  Or does this mean pushes to pull requests?  Please either explain or fix.

Reply:
Yup, see latest commit. Had 'pull' in there because the CI build wouldn't start otherwise. I think it has to do with where the PR is done. There are no github-actions on the stdlib repo yet, so nothing gets triggered this side. Its a bit funny because with gitlab the CI is picked up from whatever branch is being merged in --- otherwise changes to the CI config are never actually _tested_. Maybe there is something in the github docs that explains this, I haven't looked further.